### PR TITLE
Bug fixes and new functions in the Fuzzy Dark Matter

### PR DESCRIPTION
**Pull request by suniverse**
Adding test particles and absorbing boundaries for the fuzzy dark matter.

### CODE REVIEW

File: src/enzo/Make.config.settings [hunk 1]
```diff
@@ -215,7 +215,7 @@
 #    aggressive    Use aggressive optimization
 #-----------------------------------------------------------------------
 
-     CONFIG_OPT = debug
+     CONFIG_OPT = high
```
Comment (gregbryan) [inline src/enzo/Make.config.settings:218]: This should definitely be reverted back to debug (default should be debug).

**Review by gregbryan — COMMENTED** [review]
Thanks Xinyu!  This looks great -- just a few commented out regions to remove (if possible) and optimization setting should be set back to debug on default.

File: src/enzo/Grid_FDMCollapse.C [hunk 2]
```diff
@@ -130,9 +136,162 @@ if(QuantumPressure){
...
+			ParticleVelocity[0][npart] = 0.0;//-2.5e6/(LengthUnits/TimeUnits)*sin(theta);
+			ParticleVelocity[1][npart] =  hmcoef*8*3.1415927;//2.0e6 /(LengthUnits/TimeUnits);//*cos(theta);
+			ParticleVelocity[2][npart] = 0.0;
+		}
+		ParticleCount--;
+		npart++;
+	}
+
+    // set particles following the FDM density
+    /*
```
[the remark this answers is not in the document]
Comment (suniverse) [reply src/enzo/Grid_FDMCollapse.C:217]: Thanks Greg, I still would like to keep this part, as it sets the test particles according to some density distribution. It could be useful for future uses.

**Comment by suniverse** [comment]
There seems a CircleCI problem during the downloading of dependencies. And it caused the tests to fail.

**Review by gregbryan — APPROVED** [review]

**Comment by gregbryan** [comment]
Thanks Xinyu. The changes look good -- I have another PR pending with changes that will fix the CircleCI issues. Once that gets accepted, you can bring in those changes (very minor) and it should solve the testing issues here.

**Comment by jwise77** [comment]
@suniverse I fixed the testing in #190, and I merged in these changes to run the test suite on this PR. But I couldn't push the tags to your PR. Could you issue the command `git fetch https://github.com/jwise77/enzo-dev --tags` and then `git push` to bring in the new gold-standard-v15 tag for testing?

**Comment by jwise77** [comment]
Thanks!

**Comment by suniverse** [comment]
Hi John,

I tried to follow your instructions to fetch the tags. However, there is still an error in the test
error: pathspec 'gold-standard-v15' did not match any file(s) known to git.
Is there something wrong with I did?
Besides, I saw many changes in the run files. Are they supposed to be there?

Best
Xinyu

**Comment by jwise77** [comment]
Could you try doing "git push --tags" ?

**Comment by jwise77** [comment]
I see that didn't work either. I'm testing a new solution, and I'll push a change soon to see if the tags are fetched with it.

**Comment by jwise77** [comment]
Can you pull and push the tags again? I had to move the `gold-standard-v15` to get the parallel testing to work. I think I've fixed it. It's working in PR #188.

**Comment by jwise77** [comment]
The tag wasn't updated in your branch. Could you force pull and push? The `gold-standard-v15` should be on commit `39262f4f`.
```
git fetch  --tags --force https://github.com/jwise77/enzo-dev
git push --tags --force
```
Thanks!

**Comment by jwise77** [comment]
It passes! Thanks for working with me to get the testing incorporated into your PR. I'll review it now.

**Comment by suniverse** [comment]
Thanks John.

File: src/enzo/Make.mach.niagara [hunk 3]
```diff
@@ -0,0 +1,127 @@
+#=======================================================================
+#
+# FILE:        Make.mach.niagara
+#
+# DESCRIPTION: Makefile settings for Niagara cluster
```
Comment (jwise77) [inline src/enzo/Make.mach.niagara:5]: Add where this is located. Scinet? I would actually rename this file to include the location, like `Make.mach.scinet-niagara`

File: src/enzo/Make.mach.niagara [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+#
+# AUTHOR:      Matt Dutcher
+#
+# DATE:        2019-05-07
+#
+#=======================================================================
+
+MACH_TEXT  = Niagara
+MACH_VALID = 1
+MACH_FILE  = Make.mach.niagara
```
Comment (jwise77) [inline src/enzo/Make.mach.niagara:15]:
```suggestion
MACH_FILE  = Make.mach.scinet-niagara
```

**Review by jwise77 — CHANGES_REQUESTED** [review]
I had several minor comments and suggestions. Otherwise, looks good!

**Review by jwise77 — APPROVED** [review]

**Comment by jwise77** [comment]
Thanks for making these changes. Remove `Make.mach.niagara` since it's duplicated in `Make.mach.scinet-niagara`, and I'll merge!
